doc/classes/@GlobalScope.xml

```diff
@@ -457,6 +457,7 @@
 				    rotation = lerp_angle(min_angle, max_angle, elapsed)
 				    elapsed += delta
 				[/codeblock]
+				[b]Note:[/b] This method lerps through the shortest path between [code]from[/code] and [code]to[/code]. However, when these two angles are 180 degrees apart, it is arbitrary which way they lerp. In this particular implementation it favors lerping through the negative side. This choice is an implementation detail and may change without notice.
```

Comment:
>when these two angles are 180 degrees apart

Note that these angles are in radians so they can't really be exactly 180 degrees apart (180 degrees = PI radians and PI is irrational). Meaning in such cases the actual difference is always a little smaller/bigger than 180 degrees (by some epsilon).

>In this particular implementation it favors lerping through the negative side.

No idea what do you mean by "_through the negative side_", it's not clear. I'd say it can lerp either clockwise or counter-clockwise. And I think whether it goes clockwise/counter-clockwise depends of the error I've mentioned earlier (whether the actual angle difference is bigger/smaller than 180 degrees).

Reply:
I think this simple snippet shows it's really all about the precision and the behavior depends on the specific use case:
```
for i in range(50):
	var from := i * PI
	var to := (i + 1) * PI
	var weight := 0.5
	var result := lerp_angle(from, to, weight)
	var direction := "CLOCKWISE" if result > from else "COUNTER-CLOCKWISE"
	print("i=%-2d %-20s lerp_angle(%f, %f, %f) = %f" % [i, direction, from, to, weight, result])
```
Output (clockwise/counter-clockwise according to Godot's 2D coordinate system (Y down)):
```
i=0  COUNTER-CLOCKWISE    lerp_angle(0.000000, 3.141593, 0.500000) = -1.570796
i=1  COUNTER-CLOCKWISE    lerp_angle(3.141593, 6.283185, 0.500000) = 1.570796
i=2  COUNTER-CLOCKWISE    lerp_angle(6.283185, 9.424778, 0.500000) = 4.712389
i=3  COUNTER-CLOCKWISE    lerp_angle(9.424778, 12.566371, 0.500000) = 7.853982
i=4  COUNTER-CLOCKWISE    lerp_angle(12.566371, 15.707963, 0.500000) = 10.995574
i=5  COUNTER-CLOCKWISE    lerp_angle(15.707963, 18.849556, 0.500000) = 14.137167
i=6  COUNTER-CLOCKWISE    lerp_angle(18.849556, 21.991149, 0.500000) = 17.278760
i=7  COUNTER-CLOCKWISE    lerp_angle(21.991149, 25.132741, 0.500000) = 20.420352
i=8  COUNTER-CLOCKWISE    lerp_angle(25.132741, 28.274334, 0.500000) = 23.561945
i=9  COUNTER-CLOCKWISE    lerp_angle(28.274334, 31.415927, 0.500000) = 26.703538
i=10 CLOCKWISE            lerp_angle(31.415927, 34.557519, 0.500000) = 32.986723
i=11 COUNTER-CLOCKWISE    lerp_angle(34.557519, 37.699112, 0.500000) = 32.986723
i=12 COUNTER-CLOCKWISE    lerp_angle(37.699112, 40.840704, 0.500000) = 36.128316
i=13 CLOCKWISE            lerp_angle(40.840704, 43.982297, 0.500000) = 42.411501
i=14 CLOCKWISE            lerp_angle(43.982297, 47.123890, 0.500000) = 45.553093
i=15 COUNTER-CLOCKWISE    lerp_angle(47.123890, 50.265482, 0.500000) = 45.553093
i=16 COUNTER-CLOCKWISE    lerp_angle(50.265482, 53.407075, 0.500000) = 48.694686
i=17 CLOCKWISE            lerp_angle(53.407075, 56.548668, 0.500000) = 54.977871
i=18 CLOCKWISE            lerp_angle(56.548668, 59.690260, 0.500000) = 58.119464
i=19 COUNTER-CLOCKWISE    lerp_angle(59.690260, 62.831853, 0.500000) = 58.119464
i=20 COUNTER-CLOCKWISE    lerp_angle(62.831853, 65.973446, 0.500000) = 61.261057
i=21 CLOCKWISE            lerp_angle(65.973446, 69.115038, 0.500000) = 67.544242
i=22 COUNTER-CLOCKWISE    lerp_angle(69.115038, 72.256631, 0.500000) = 67.544242
i=23 COUNTER-CLOCKWISE    lerp_angle(72.256631, 75.398224, 0.500000) = 70.685835
i=24 COUNTER-CLOCKWISE    lerp_angle(75.398224, 78.539816, 0.500000) = 73.827427
i=25 COUNTER-CLOCKWISE    lerp_angle(78.539816, 81.681409, 0.500000) = 76.969020
i=26 CLOCKWISE            lerp_angle(81.681409, 84.823002, 0.500000) = 83.252205
i=27 COUNTER-CLOCKWISE    lerp_angle(84.823002, 87.964594, 0.500000) = 83.252205
i=28 COUNTER-CLOCKWISE    lerp_angle(87.964594, 91.106187, 0.500000) = 86.393798
i=29 CLOCKWISE            lerp_angle(91.106187, 94.247780, 0.500000) = 92.676983
i=30 COUNTER-CLOCKWISE    lerp_angle(94.247780, 97.389372, 0.500000) = 92.676983
i=31 COUNTER-CLOCKWISE    lerp_angle(97.389372, 100.530965, 0.500000) = 95.818576
i=32 COUNTER-CLOCKWISE    lerp_angle(100.530965, 103.672558, 0.500000) = 98.960169
i=33 COUNTER-CLOCKWISE    lerp_angle(103.672558, 106.814150, 0.500000) = 102.101761
i=34 CLOCKWISE            lerp_angle(106.814150, 109.955743, 0.500000) = 108.384947
i=35 COUNTER-CLOCKWISE    lerp_angle(109.955743, 113.097336, 0.500000) = 108.384947
i=36 COUNTER-CLOCKWISE    lerp_angle(113.097336, 116.238928, 0.500000) = 111.526539
i=37 CLOCKWISE            lerp_angle(116.238928, 119.380521, 0.500000) = 117.809725
i=38 COUNTER-CLOCKWISE    lerp_angle(119.380521, 122.522113, 0.500000) = 117.809725
i=39 COUNTER-CLOCKWISE    lerp_angle(122.522113, 125.663706, 0.500000) = 120.951317
i=40 CLOCKWISE            lerp_angle(125.663706, 128.805299, 0.500000) = 127.234502
i=41 COUNTER-CLOCKWISE    lerp_angle(128.805299, 131.946891, 0.500000) = 127.234502
i=42 CLOCKWISE            lerp_angle(131.946891, 135.088484, 0.500000) = 133.517688
i=43 CLOCKWISE            lerp_angle(135.088484, 138.230077, 0.500000) = 136.659280
i=44 COUNTER-CLOCKWISE    lerp_angle(138.230077, 141.371669, 0.500000) = 136.659280
i=45 CLOCKWISE            lerp_angle(141.371669, 144.513262, 0.500000) = 142.942466
i=46 COUNTER-CLOCKWISE    lerp_angle(144.513262, 147.654855, 0.500000) = 142.942466
i=47 CLOCKWISE            lerp_angle(147.654855, 150.796447, 0.500000) = 149.225651
i=48 CLOCKWISE            lerp_angle(150.796447, 153.938040, 0.500000) = 152.367244
i=49 COUNTER-CLOCKWISE    lerp_angle(153.938040, 157.079633, 0.500000) = 152.367244
```

Reply:
> So it's not so much an implementation details as unpredictable.

I think it's predictable if the user dig into how IEEE 754 float point works, but not worth the effort :stuck_out_tongue_closed_eyes: